test/hotspot/jtreg/compiler/intrinsics/TestArraysHashCode.java

```diff
@@ -25,43 +25,66 @@
  * @test
  * @bug 8301093
  * @summary Verify failure to intrinsify does not pollute control flow
- * @modules java.base/jdk.internal.util
+ * @modules java.base/jdk.internal.util:+open
  *
  * @run main/othervm -Xbatch -XX:-TieredCompilation compiler.intrinsics.TestArraysHashCode
  */
 
 package compiler.intrinsics;
 
-import jdk.internal.util.ArraysSupport;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
 
 public class TestArraysHashCode {
 
+    private static final Method vectorizedHashCode;
+    private static final int T_BOOLEAN;
+
+    static {
+        try {
+            var arraysSupport = Class.forName("jdk.internal.util.ArraysSupport");
+            vectorizedHashCode = arraysSupport.getDeclaredMethod("vectorizedHashCode",
+                    Object.class, int.class, int.class, int.class, int.class);
+            vectorizedHashCode.setAccessible(true);
+            Field f = arraysSupport.getDeclaredField("T_BOOLEAN");
+            f.setAccessible(true);
+            T_BOOLEAN = f.getInt(null);
+        } catch (Throwable e) {
+            throw new RuntimeException(e);
+        }
+    }
+
     static int type;
     static byte[] bytes;
 
-    public static void main(String[] args) {
+    public static void main(String[] args)
+            throws InvocationTargetException, IllegalAccessException {
         // read
         bytes = new byte[256];
-        type = ArraysSupport.T_BOOLEAN;
+        type = T_BOOLEAN;
         testIntrinsicWithConstantType();
         testIntrinsicWithNonConstantType();
     }
 
-    private static void testIntrinsicWithConstantType() {
+    private static void testIntrinsicWithConstantType()
+            throws InvocationTargetException, IllegalAccessException {
         for (int i = 0; i < 20_000; i++) {
-            testIntrinsic(bytes, ArraysSupport.T_BOOLEAN);
+            testIntrinsic(bytes, T_BOOLEAN);
         }
     }
 
     // ok, but shouldn't be intrinsified due the non-constant type
-    private static void testIntrinsicWithNonConstantType() {
-        type = ArraysSupport.T_BOOLEAN;
+    private static void testIntrinsicWithNonConstantType()
+            throws InvocationTargetException, IllegalAccessException {
+        type = T_BOOLEAN;
         for (int i = 0; i < 20_000; i++) {
             testIntrinsic(bytes, type);
         }
     }
 
-    private static int testIntrinsic(byte[] bytes, int type) {
-        return ArraysSupport.vectorizedHashCode(bytes, 0, 256, 1, type);
+    private static int testIntrinsic(byte[] bytes, int type)
+            throws InvocationTargetException, IllegalAccessException {
+        return (int) vectorizedHashCode.invoke(null, bytes, 0, 256, 1, type);
```

Comment:
Better to just call `hashCodeOfUnsigned` here I think.

The test for the non-constant type could be dropped. That is no longer a part of the 'API' of `ArraySupport`. It looks like the intrinsic bails out when the basic type is not constant any ways: https://github.com/openjdk/jdk/blob/master/src/hotspot/share/opto/library_call.cpp#L6401-L6404

Reply:
The non-constant test was added because that very bailout caused a crash. The other test is actually less interesting since it'll likely be covered indirectly by regular use. But as we are hiding these away this gets ever more obscure and perhaps the test could be dropped entirely.

Reply:
@cl4es, do you want me to delete that test file altogether?

Reply:
I thought you verified that the non-constant type test still provoke a crash (on x86) if you back out the code changes in https://github.com/openjdk/jdk/commit/969f6a37e4649079c7acea1952f5537fd9ba2f0a ? If so that test is still somewhat useful to guard against future coding mistakes by verifying that the bail out doesn't mess things up. The constant type tests have less utility, perhaps. I'd keep it as is unless there's a strong desire to reduce test runtime (these should be pretty quick).

Reply:
I did verify it. Sorry, momentary lapse of the ability to reason. I'll integrate this PR very shortly then. Thanks.